Add glossary for injection and gateway (#9733)
istio/istio.io: content/en/docs/reference/glossary/gateway.md

---
title: Gateway
test: n/a
---

> A gateway refers to standalone Istio proxies deployed at the edge of the mesh.

Reviewer: ```suggestion
A gateway is a standalone Istio proxy deployed at the edge of the mesh.
```

> These are used to route traffic [into](/docs/tasks/traffic-management/ingress/) or [out of](/docs/tasks/traffic-management/egress/) the mesh.

Reviewer: Do you think we should also mention Gateway CR?
```suggestion
Gateways are used to route traffic [into](/docs/tasks/traffic-management/ingress/) or [out of](/docs/tasks/traffic-management/egress/) the mesh.

An Istio `Gateway` CR is used to configure the exposed ports of a gateway deployment.
```
